### PR TITLE
ddl_test: fix the wrong test #9610

### PR DESCRIPTION
**Pull request by winkyao**
Cherry-pick from #9610

### CODE REVIEW

File: ddl/ddl_worker_test.go [hunk 1]
```diff
@@ -303,7 +303,7 @@ func checkCancelState(txn kv.Transaction, job *model.Job, test *testCancelJob) e
 	// If the action is adding index and the state is writing reorganization, it wants to test the case of cancelling the job when backfilling indexes.
 	// When the job satisfies this case of addIndexFirstReorg, the worker hasn't started to backfill indexes.
 	if test.cancelState == job.SchemaState && !addIndexFirstReorg {
-		if job.SchemaState == model.StateNone && job.State != model.JobStateDone && job.Type != model.ActionCreateTable && job.Type != model.ActionCreateSchema && job.Type != model.ActionRebaseAutoID {
+		if job.SchemaState == model.StateNone && job.State != model.JobStateDone && job.Type != model.ActionCreateTable && job.Type != model.ActionCreateSchema && job.Type != model.ActionRebaseAutoID && job.Type != model.ActionShardRowID {
```
Comment (xiekeyi98) [inline ddl/ddl_worker_test.go]: Why this change is  different from https://github.com/pingcap/tidb/pull/9610?

Comment (crazycs520) [reply ddl/ddl_worker_test.go]: you can see https://github.com/pingcap/tidb/pull/9610/files#diff-30da5c417c2d55380ca28e19605383eeL342.
